File: week07/total_words.pl
#!/usr/bin/perl
use strict;
use warnings;

use English;

sub findWords {
    my ($input) = @ARG;

    my @words;
    while (my $line = <$input>) {
        push @words, grep(/./, split(/[^a-zA-Z]/, $line));
    }

    return @words;
}

if (!caller) {
    printf "%d words\n", scalar findWords \*STDIN;
}

1;

File: week07/total_words.py
#!/usr/bin/python3

import io, re, sys

__all__ = ["findWords"]

def findWords(iterable):
    for line in iterable:
        for word in filter(lambda w: w, re.split("[^a-zA-Z]+", line)):
            yield word

if __name__ == "__main__":
    print(len(list(findWords(io.TextIOWrapper(sys.stdin.buffer, encoding = "utf-8")))), "words")
